src/front/wgsl/mod.rs

```diff
@@ -2901,6 +2910,23 @@ impl Parser {
         Ok((members, span))
     }
 
+    fn parse_matrix_scalar_type<'a>(
+        &mut self,
+        lexer: &mut Lexer<'a>,
+        columns: crate::VectorSize,
+        rows: crate::VectorSize,
+    ) -> Result<crate::TypeInner, Error<'a>> {
+        let (kind, width, span) = lexer.next_scalar_generic_with_span()?;
+        match (kind, width) {
+            (crate::ScalarKind::Float, 4) => Ok(crate::TypeInner::Matrix {
```

Comment:
We shouldn't check the width here since it will be checked by the validation code where a width of 8 might be allowed.

https://github.com/gfx-rs/naga/blob/ab2806e05fbd69a502b4b25a85f99ae4d3e82278/src/valid/type.rs#L205-L208

Reply:
@grovesNL I think you missed this one

Reply:
Oops, thanks for catching this! Should be fixed now